test-app/backend/test/validators/js/zod/zod.test.ts

```diff
@@ -0,0 +1,26 @@
+import { ZodValidator } from 'src/validators/js/zod/zod';
+import { z } from 'zod';
+
+const TEST_SCHEMA = z.number().min(1);
+
+describe('ZodValidator', () => {
+  function setup() {
+    return {
+      validator: new ZodValidator(),
+    };
+  }
+
+  test('validation with valid data should pass', async () => {
+    const { validator } = setup();
+
+    const value = await validator.runValidate(8, { schema: TEST_SCHEMA });
+
+    expect(value).toBe(8);
+  });
+
+  test('validation with invalid data should fail', async () => {
+    const { validator } = setup();
+
+    await expect(validator.runValidate(0, { schema: TEST_SCHEMA })).rejects.toEqual(expect.any(z.ZodError));
```

Comment:
Just curious about the benefit of testing zod validation logic? Should we treat it as a black box instead?

Reply:
The actual Zod validation is tested a lot. I don't really think it's worth removing the validation right now just because it's cumbersome. This unit test is actually just meant to make sure the runValidate has expected behavior in general.